x-pack/plugins/infra/public/utils/logs_overview_fetchers.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { InfraClientCoreSetup } from '../types';
+import { LogsFetchDataResponse } from '../../../observability/public';
+
+export function getLogsHasDataFetcher(getStartServices: InfraClientCoreSetup['getStartServices']) {
+  return async () => {
+    // if you need the data plugin, this is how you get it
+    const [, startPlugins] = await getStartServices();
+    const { data } = startPlugins;
+
+    // if you need a core dep, we need to pass in more than just getStartServices
+
+    // perform query
+    return true;
+  };
+}
+
+export function getLogsOverviewDataFetcher(
+  getStartServices: InfraClientCoreSetup['getStartServices']
+) {
+  return async (): Promise<LogsFetchDataResponse> => {
+    // if you need the data plugin, this is how you get it
+    const [, startPlugins] = await getStartServices();
+    const { data } = startPlugins;
+
+    // if you need a core dep, we need to pass in more than just getStartServices
+
+    // perform query
+    return {
+      title: 'Log rate',
+      appLink: 'TBD', // TODO: what format should this be in, relative I assume?
```

Comment:
i think for the sake of simplicity observability plugin should handle the relative part while generating link.